### ISSUE TITLE
Request for comments: use a command mode for opening and closing gnome-control-center

### ISSUE THREAD

**Issue by ferdinandyb** [regolith-linux/regolith-i3-gaps-config#6]
**Problems**
1. Current bindings for `gnome-control-center` take up many of the "most valuable" $mod + single key bindings
2. `gnome-control-center` behaviour is not very tiling wm friendly, because when it is losing width, it starts to decrease the area for the settings themselves, while keeping the left hand side tabs the same size.

**Proposed solution**: Create a command mode for handling the control center as a floating window.

I propose to move all the keybinds into a control mode. Entering control mode with `$mod+c` would open the gnome-control-center, if it is already open fetch it from the scratchpad as a floating window (potentially, it could also make it full screen?). Closing the control mode with the same key combo puts the gnome-control-center on the scratchpad hiding it. In case you need to type something, you will need to use Esc to close the mode and e.g. type the wifi password and then use `$mod+Ctrl+m` to move the control center to the scratchpad (or just kill it since starting control mode will spawn a new one anyway). In control mode, single key presses will take you to various settings pages of the control center.

Note, that the bindings that are not the control mode in the below code should probably be thought about, since I set those binds with the original bindings of the control center still active, now with the new free binding place there might be some better combos (although it might be good to leave them free for users to bind some functionalities they use more often, because IMHO the scratchpad is only useful if you can set it to fetch specific apps, like here the control center, or the music player or a terminal etc.)

```
# toggle tiling / floating
bindsym $mod+Ctrl+f floating toggle
bindsym $mod+Ctrl+m move to scratchpad
bindsym $mod+Ctrl+a scratchpad show
bindsym $mod+Ctrl+x floating disable
bindsym $mod+m move workspace to output right


# change focus between tiling / floating windows
bindsym $mod+Ctrl+t focus mode_toggle

# Use Mouse+$mod to drag floating windows to their wanted position
floating_modifier $mod
floating_maximum_size -1 x -1

mode "Control Center" {

      # shortcuts for common system configuration tasks
      bindsym w exec gnome-control-center wifi
      bindsym b exec gnome-control-center bluetooth
      bindsym d exec gnome-control-center display
      bindsym n exec gnome-control-center network
      bindsym p exec gnome-control-center power
      bindsym s exec gnome-control-center sound
        
      bindsym Escape mode "default"
      bindsym $mod+c mode "default", [class=Gnome-control-center] move to scratchpad
}

bindsym $mod+c mode "Control Center",exec --no-startup-id gnome-control-center, [class=Gnome-control-center] move to scratchpad, scratchpad show, resize set width 96 ppt height 95 ppt, move position center
```

See here: https://github.com/priestoferis/regolith-i3-gaps-config/blob/6c63bacf09dc2aadd5cc6d3972cd3e517cfce6de/config#L192

Any thoughts about this approach?

**Comment by thezeroalpha** [comment]
Good idea, I like this. I also agree that a lot of the convenient $mod bindings are taken up by gnome-control-center; a 'control mode' seems like an elegant solution to this.

**Comment by ploum** [comment]
I would be curious to test it. I agree with the problems.

**Comment by ferdinandyb** [comment]
Do let me know, if you've managed to test it!

**Comment by ferdinandyb** [comment]
I've been also thinking, that IF we manage to free up some keybinds, maybe binding a file browser would be a nice addition to the terminal and browser duo as I think these are the most used things (although for myself I'm thinking on throwing it out on a scratchpad as well, but anyway).

**Comment by kgilmer** [comment]
I am not sure about the scratchpad part, only because I haven't had time to play around with it, otherwise all proposed changes I like and want to add to the Regolith R1.3 release.  Thanks for the great ideas, providing example code, and sound reasoning @priestoferis !

**Comment by ferdinandyb** [comment]
I have one issue with my scratchpad solution now, is that when you get a new dialog (bluetooth connect or wifi password), and you exit the mode (sending everything to the scratchpad) bringing it up again will put the smaller float BEHIND the bigger float, which is super annoying. I also noticed that to connect to a new wifi, I have to go into the network settings, clicking on the wifi on control center doesn't bring up any password dialog (but I think this is unrelated to anything here). I need to do some testing.

**Comment by kgilmer** [comment]
Can we maybe take the floating and i3 mode if the scratchpad needs some more time or do you think it's all-or-nothing?

**Comment by ferdinandyb** [comment]
I'm not really sure what you mean :) But, I did some testing and came up with a slight modification, that works the same way (sending to scratchpad), but when in front, it tiles. A clear benefit of this is that in case you "mess up" and send the control center to the scratchpad with another dialog open, and then bring both back, you will see both, or only send the "main" control center window and not the option floats. The best would be if we could capture the id of the main window and later use that, but that doesn't seem possible in a simple config file (although we could call an external script to handle this, it should not be hard to solve with the python bindings).

So for now I suggest adding this, instead of the control mode I pasted before:

```
mode "Control Center" {

      # shortcuts for common system configuration tasks
      bindsym w exec gnome-control-center wifi
      bindsym b exec gnome-control-center bluetooth
      bindsym d exec gnome-control-center display
      bindsym n exec gnome-control-center network
      bindsym p exec gnome-control-center power
      bindsym s exec gnome-control-center sound
        
      bindsym Escape mode "default"
      bindsym $mod+c mode "default", [class=Gnome-control-center] move to scratchpad
}

bindsym $mod+c mode "Control Center", exec gnome-control-center, [class=Gnome-control-center] move to scratchpad, scratchpad show, floating disable
```

What do you think about later $mod+c changing mode and calling a script to handle where the windows go?

**Comment by kgilmer** [comment]
After some experimentation, I find that the scratchpad aspect of this to be confusing and finicky.  Also (perhaps there is something in my local configuration causing this) the behavior seems to change between the first time I invoke `gnome-control-center` and successive tries. 

Perhaps it's because I personally don't use scratchpads, but I find the fact that the gnome-control-center continues to run after it's been dismissed from the focused workspace to be confusing and non obvious.  When trying to launch it later via Rofi, nothing seemingly happens (because it's already loaded in a scratchpad) which may be difficult for new users.

Regarding the 'control center' mode, I would expect that the control center exits when I exit the mode, but it does not.  I have to explicitly close it.  Or, w/ the scratchpad, navigate there and close it myself.  To me this makes the mode style less compelling than just launching the app.

I suggest going with a (IMHO) simpler approach;
1. have gnome-control-center launch in floating mode 
2. remove all of the keybindings except c (launch settings), d (load display), and w (load wifi panel), my reasoning being that changing display and wifi settings is very common with laptops. 
3. Do not use scratchpads or modes.
4. Add binding to Files (Nautilus) with shift-n.

In essence, this:
```
# shortcuts for common system configuration tasks
bindsym $mod+c exec gnome-control-center
bindsym $mod+d exec gnome-control-center display
bindsym $mod+w exec gnome-control-center wifi
bindsym $mod+Shift+n exec nautilus
...

floating_maximum_size -1 x -1
for_window [class="Gnome-control-center"] floating enable
```

It could be that I'm missing something with the resizing but I believe the control panel will provide a size itself and so we do not need to resize it to a certain percentage of the screen.  Indeed, on my 32" 4K monitor, with the previous settings it takes up almost all of the screen which is way too much to be practical. 

I will package this in the new unstable PPA (ppa:regolith-linux/unstable) to give others a chance to try it out.

**Comment by kgilmer** [comment]
This has been shipped in 1.3.
